### PR TITLE
ADD: get_airnow functionality

### PR DESCRIPTION
**Pull request by jhemedin**
Added function to pull the AirNow data from the AirNow API. Two of the functions just pull the data and display them on screen and the last one will download a file. Working on adding the option to download files with the other functions that just print to the screen.

Need to figure out how to unit test since a users private token is needed to access the data.

### CODE REVIEW

**Comment by AdamTheisen** [comment]
@mgrover1 would you want to tab a stab at helping on this PR?  This is a new capability to work with the EPA's airnow portal to pull data. https://docs.airnowapi.org/.  The main thing we need is to set up unit tests for this but the portal requires a username/token so we were going to follow like the testing of the ARM download data test and have environment variables set on TRAVIS-CI with a username/token.  I'm fine setting up an account and using my credentials there but also open if you have other ideas.

**Comment by mgrover1** [comment]
@AdamTheisen Sure! I can take a look at this!

**Comment by AdamTheisen** [comment]
I've made a fair amount of updates here.  

1. I've simplified get_AirNow_forecast and get_AirNow_obs to remove the format as we are returning an xarray dataset anyways
2. Revamped get_AriNow to return an xarray object instead of downloading files to be consistent with the others.  I also simplified and renamed to get_AirNow_bounded_obs
3. Added tests to test_discovery and in doing so found a failing test from cropscape that I fixed.

**Comment by zssherman** [comment]
Looks good, only thing failing at the moment is coveralls, but with new code, i kinda expect it. There are unittests you added so once the rest passes, think this would be good to merge.

**Review by zssherman — APPROVED** [review]

**Comment by AdamTheisen** [comment]
Given that these tests will use secrets, it won't run during the PR testing.  I've updated the coverage failure to 1.5% to account for this.

File: act/discovery/get_airnow.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+def get_AirNow_forecast(token, date, zipcode=None, latlon=None, distance=25):
+    """
+    This tool will get current or historical AQI values and categories for a
+    reporting area by either Zip code or Lat/Lon coordinate.
+
+    Parameters
+    ----------
+    token : str
+        The access token for accesing the AirNowAPI web server
```
Comment (mgrover1) [inline act/discovery/get_airnow.py:14]: One small suggestion - can you add the link here to the AirNowAPI? Just thinking about the user trying to find where to get their login.

File: act/discovery/get_airnow.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
+import pandas as pd
+import numpy as np
+import xarray as xr
+
+
+def get_AirNow_forecast(token, date, zipcode=None, latlon=None, distance=25):
```
Comment (mgrover1) [inline act/discovery/get_airnow.py:6]: Can you make this function all lower-case?

File: act/discovery/get_airnow.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+    df = pd.read_csv(url)
+
+    # converting to xarray object
+    df = df.to_xarray()
+
+    return df
+
+
+def get_AirNow_obs(token, date=None, zipcode=None, latlon=None, distance=25):
```
Comment (mgrover1) [inline act/discovery/get_airnow.py:68]: Same here - can you make this all lowercase?

File: act/discovery/get_airnow.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    """
+    This tool will get current or historical observed AQI values and categories for a
+    reporting area by either Zip code or Lat/Lon coordinate.
+
+    Parameters
+    ----------
+    token : str
+        The access token for accesing the AirNowAPI web server
```
Comment (mgrover1) [inline act/discovery/get_airnow.py:76]: And add the link here too.

File: act/discovery/get_airnow.py [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+
+    df = pd.read_csv(url)
+
+    # converting to xarray
+    df = df.to_xarray()
+
+    return df
+
+
+def get_AirNow_bounded_obs(token, start_date, end_date, latlon_bnds, parameters='OZONE,PM25', data_type='B',
```
Comment (mgrover1) [inline act/discovery/get_airnow.py:144]: Lowercase here

File: act/discovery/get_airnow.py [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+                           mon_type=0):
+    """
+    Get AQI values or data concentrations for a specific date and time range and set of
+    parameters within a geographic area of intrest
+
+    Parameters
+    ----------
+    token : str
```
Comment (mgrover1) [inline act/discovery/get_airnow.py:152]: And link in the in the docstring

**Review by mgrover1 — CHANGES_REQUESTED** [review]
Just a few suggestions @AdamTheisen

**Comment by AdamTheisen** [comment]
@mgrover1 these latest updates should take care of your changes that were requested.  Added in a link to each function description and updated the functions to be lower case

**Review by mgrover1 — APPROVED** [review]
Great! Thanks for making those changes!

**Comment by mgrover1** [comment]
Once the tests pass, we can merge away 👍

**Pull request merged**